src/Microsoft.AspNetCore.Mvc.Core/Authorization/AuthorizeFilter.cs

```diff
@@ -46,11 +77,17 @@ public AuthorizeFilter(AuthorizationPolicy policy)
                 throw new ArgumentNullException(nameof(context));
             }
 
+            var effectivePolicy = Policy ?? await AuthorizationPolicy.CombineAsync(PolicyProvider, AuthorizeData);
```

Comment:
Is there benefit to caching `EffectivePolicy`? Doesn't seem like `CombineAsync` has per invocation data to work with.